pkg/registry/core/pod/storage/eviction.go

```diff
@@ -130,13 +130,46 @@ func (r *EvictionREST) Create(ctx context.Context, name string, obj runtime.Obje
 
 	// Evicting a terminal pod should result in direct deletion of pod as it already caused disruption by the time we are evicting.
 	// There is no need to check for pdb.
-	if pod.Status.Phase == api.PodSucceeded || pod.Status.Phase == api.PodFailed {
-		_, _, err = r.store.Delete(ctx, eviction.Name, rest.ValidateAllObjectFunc, deletionOptions)
-		if err != nil {
-			return nil, err
+	if canIgnorePDB(pod) {
+		if shouldEnforceResourceVersion(pod) {
+			// Set deletionOptions.Preconditions.ResourceVersion to ensure we're not
+			// racing with another PDB-impacting process elsewhere.
+			if deletionOptions.Preconditions == nil {
+				deletionOptions.Preconditions = &metav1.Preconditions{}
+			}
+			deletionOptions.Preconditions.ResourceVersion = &pod.ResourceVersion
```

Comment:
You're modifying `deletionOptions.Preconditions.ResourceVersion` here in a way that is correct for the canIgnorePDB case, but is not correct if we end up in the continueToPDBs case. In that scenario, we want the original deletionOptions to be used. Need a test case to cover this scenario:
1. eviction request is received with no resourceVersion precondition
2. pod at rv=1 is pending, so canIgnorePDBs is true
3. deletionOptions gets a rv=1 precondition added
4. initial eviction attempt gets a conflict error
5. next pod get returns non-pending pod at rv=2, so canIgnorePDBs is false, so we continue to PDBs
6. PDB allows eviction
7. Pod is deleted successfully

Currently, that scenario would fail in step 7 with a conflict error because of the leftover rv=1 precondition